### PR TITLE
Use Flexbox for overall layout/grids

### PR DESCRIPTION
**Pull request by rschlaefli**
GENERAL
- Use flexbox for all pages of the application (replacing all semantic grids)
- Add responsive overall layout for `Navbar`, `Sidebar` and pages
- Remove footer (we won't really need this inside the application)
- Place the action button at a fixed position and restructure components as needed to fit

MISC
- Remove padding inherent to `Question` container (should be controlled from the outside)
- Pass the `key` down in `QuestionList`
- Add *loading* state for running session page to prevent errors

### CODE REVIEW

File: src/components/layouts/TeacherLayout.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import { Grid } from 'semantic-ui-react'
+// import { Grid } from 'semantic-ui-react'
```
Comment (rschlaefli) [inline src/components/layouts/TeacherLayout.js]: No longer needed

File: src/pages/questions/index.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ class Index extends Component {
 
   // handling the state of the new course button
   handleActiveNewButton = () => {
+    console.dir('hello world')
```
Comment (rschlaefli) [inline src/pages/questions/index.js]: Remove logging (the button does work).

File: src/pages/questions/index.js [hunk 3]
```diff
@@ -74,48 +75,31 @@ class Index extends Component {
...
+            <Grid.Column width="2">
               <TagList activeTags={this.state.activeTags} handleTagClick={this.handleTagClick} />
             </Grid.Column>
-            <Grid.Column stretched width="12">
+            <Grid.Column tablet="14" computer="12">
               <QuestionList />
             </Grid.Column>
-            <Grid.Column stretched textAlign="right" verticalAlign="bottom" width="2">
-              {/* TODO correct position of button */}
-              {this.state.activeNewButton &&
```
Comment (rschlaefli) [inline src/pages/questions/index.js]: Will need to readd this menu and the button transforms.

File: src/components/common/navbar/Navbar.js [hunk 4]
```diff
@@ -45,54 +45,88 @@ class Navbar extends Component {
...
-                <Icon name="sidebar" />
-              </Menu.Item>
-              <Menu.Header as="h1" className="navbarTitle" content={title} />
-            </Menu.Menu>
+        <div className="sideArea">
+          <Menu borderless className="noBorder">
+            <Menu.Item icon active={sidebarVisible} name="sidebar" onClick={handleSidebarToggle}>
+              <Icon name="sidebar" />
+            </Menu.Item>
+            <h1>
```
Comment (pascalzehnder) [inline src/components/common/navbar/Navbar.js]: Do you wanna have such small statements padded generally? Don't you think one line would be enough? :)

File: src/components/common/navbar/Navbar.js [hunk 5]
```diff
@@ -45,54 +45,88 @@ class Navbar extends Component {
...
-              font-size: 1.3rem;
-              margin-left: 1rem;
-            }
+        <style jsx>{`
+          .navbar {
+            display: flex;
+            align-items: center;
+            flex-flow: row wrap;
+            justify-content: space-between;
+
```
Comment (pascalzehnder) [inline src/components/common/navbar/Navbar.js]: Spacing!

**Review by pascalzehnder — COMMENTED** [review]
Well done!

File: src/components/common/navbar/Navbar.js (hunk 4 above)
Comment (rschlaefli) [inline src/components/common/navbar/Navbar.js]: Prettier does this automatically :) You should install it too, you know.

File: src/components/common/navbar/Navbar.js (hunk 5 above)
Comment (rschlaefli) [inline src/components/common/navbar/Navbar.js]: On purpose!

@pascalzehnder I'm leaving a space between rules concerning flexbox and other rules.

**Review by pascalzehnder — DISMISSED** [review]
Go!

**Review by pascalzehnder — APPROVED** [review]

**Pull request merged**